src/Session.php

```diff
@@ -1683,6 +1683,10 @@ public static function cleanIDORTokens()
             foreach ($_SESSION['glpiidortokens'] as $footprint => $token) {
                 if ($token['expires'] < $now) {
                     unset($_SESSION['glpiidortokens'][$footprint]);
+                } else {
+                    // Token are stored from oldest to youngest
+                    // If this token is still valid, all the remaining token are valid too
+                    break;
                 }
             }
```

Comment:
I do not know how PHP is handling write operations on `$_SESSION`. If each variable set/unset triggers a write operation on disk, then the better solution is probably to use `array_filter` in order to clean tokens in a unique writing operation.